app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -92,8 +123,16 @@ private void setupItemListener(QuerySnapshot querySnapshots, FirebaseFirestoreEx
             for (QueryDocumentSnapshot doc: querySnapshots) {
                 Map<String, Object> data = new HashMap<>(doc.getData());
                 itemList.add(new Item(doc.getId(), data));
-                itemAdapter.notifyDataSetChanged();
             }
+//                    Every time the data list is updated, all the checkboxes are unselected
+            for (int i=0;i<itemList.size();i++) {
```

Comment:
Two things, first do we actually need to setChecked false? I believe the snapshot listener wipes the itemList with ```itemList.clear();```. Not sure if that impacts the state in the list or not. Second, if we do really need this I am pretty sure we can avoid two for loops here by simply combining with the querySnapshots one as querySnapshots.size() = itemList.size()

Reply:
Also you could just call your unselectAllItems method

Reply:
To your first comment, it is kind of necessary since the checkboxes are part of itemAdapter/itemView, not the Item class. So say you select an item meaning you selected a checkbox on its adapter, when you delete that item, the adapter still exists. The ArrayList also gets rearranged and some other item takes the previous item's position and it appears that it is selected because the checkbox is still ticked.

So either I move this for loop to unselectAllItems or I can make a single for loop in the ItemListener as you said. Lemme know which do you prefer

Reply:
I think sticking to the unselect method is probably best. Keep things consistent